### PR TITLE
Use pip-tools for requirement tracking

### PR DESCRIPTION
**Pull request by blampe**
Kicking the tires on pip-tools, followup to #334.

@breerly @abhinav

### CODE REVIEW

File: requirements.in [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# stdlib backports, no constraints needed
+contextlib2
+
+# external deps
+crcmod>=1,<2
+tornado>=4.2,<5
+
+# tchannel deps
+thriftrw>=0.4,<2
```
Comment (abhinav) [inline requirements.in]: Can you instead list the dependencies in setup.py and change this to `-e .`?

Comment (abhinav) [inline requirements.in]: Also, you'd have to add `requirements.in` to the MANIFEST.in if you want setup.py to read it.

File: requirements.txt [hunk 2]
```diff
@@ -1,10 +1,18 @@
...
-# tchannel deps
-thriftrw>=0.4,<2
-threadloop>=1,<2
+#
+# This file is autogenerated by pip-compile
+# Make changes in requirements.in, then run this to update:
+#
+#    pip-compile --no-index requirements.in
+#
+-e file:///Users/blampe/src/tchannel-python
```
Comment (junchaowu) [inline requirements.txt]: ?

Comment (abhinav) [inline requirements.txt]: Does pip-compile not support `-e .`?

**Comment by abhinav** [comment]
Also, if we figure this out, would be nice to have a Makefile target to update all the requirements.

**Comment by abhinav** [comment]
:ok:

**Comment by breerly** [comment]
This is great - will probably want to do the same with thriftrw-python and yarpc-python.

**Comment by breerly** [comment]
@blampe can you link to the article that describes this workflow?

You might want to squash as well.

**Comment by blampe** [comment]
@breerly squashed, http://nvie.com/posts/better-package-management/

**Pull request merged**